aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -77,23 +77,32 @@
 
         if (webSocket == null) {
             try {
+                connectionFailure = null;
                 webSocket = createWebSocket();
             } catch (ApiException exception) {
                 responseListener.onError(new ApiException(
                         "Failed to create websocket for subscription",
                         exception,
                         AmplifyException.TODO_RECOVERY_SUGGESTION
                 ));
+                return null;
             }
 
             try {
                 connectionAcknowledgement.await(CONNECTION_ACKNOWLEDGEMENT_TIMEOUT, TimeUnit.SECONDS);
-            } catch (InterruptedException interruptedException) {
+            } catch (InterruptedException interruptedException) { }
+
+            if (connectionAcknowledgement.getCount() != 0) {
                 responseListener.onError(new ApiException(
                         "Subscription timed out waiting for acknowledgement",
-                        interruptedException,
                         AmplifyException.TODO_RECOVERY_SUGGESTION
                 ));
+                return null;
+            } else if (connectionFailure != null) {
+                responseListener.onError(
+                    new ApiException(connectionFailure, "Check if you are authorized to make this subscription")
```

Comment:
Since this is in response to a bug report, and is the primary intent of the PR, we should add some test to cover this scenario.

Reply:
Alright - also in running the integration tests I'm seeing a problem with a normal subscription now so trying to figure out the issue.

Reply:
It might be the race condition we were talking about earlier, where a couple of different tests try use subscriptions for the same model type, but the order of execution of the tests matters, or something?